src/IO/FileEncryptionCommon.cpp

```diff
@@ -364,54 +370,87 @@ void Encryptor::decrypt(const char * data, size_t size, char * out)
 
 void Header::read(ReadBuffer & in)
 {
-    constexpr size_t header_signature_size = std::size(kHeaderSignature) - 1;
-    char signature[std::size(kHeaderSignature)] = {};
-    in.readStrict(signature, header_signature_size);
-    if (strcmp(signature, kHeaderSignature) != 0)
+    char signature[kHeaderSignature.length() + 1] = {};
```

Comment:
One additional byte for `\0` ?

Reply:
yes, but it's not neccessary actually because `memcmp()` can be used instead.